app/components/post_list/post/unread_dot.tsx

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {View} from 'react-native';
+
+import {makeStyleSheetFromTheme} from '@utils/theme';
+
+import type ThreadModel from '@typings/database/models/servers/thread';
+
+type Props = {
+    testID: string;
+    theme: Theme;
+    thread: ThreadModel;
+};
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        badgeContainer: {
+            position: 'absolute',
+            left: 21,
+            bottom: 9,
+        },
+        unreadDot: {
+            width: 8,
+            height: 8,
+            borderRadius: 4,
+            backgroundColor: theme.sidebarTextActiveBorder,
+            alignSelf: 'center',
+            top: -6,
+            left: 4,
+        },
+    };
+});
+
+const UnreadDot = ({testID, theme, thread}: Props) => {
+    if (thread.unreadMentions || thread.unreadReplies) {
```

Comment:
you could check for this before mounting the component in the post component. and the theme you should get from context.